loky/backend/utils.py

```diff
@@ -110,3 +107,296 @@ def _recursive_terminate(pid):
+            PROCESS_QUERY_INFORMATION | PROCESS_VM_READ,
+            False, os.getpid())
+        if not hProcess:
+            raise OSError('Could not open PID %s' % os.getpid())
+
+        found_module_path = None
+        try:
+            buf_count = 256
+            needed = DWORD()
+            # Grow the buffer until it becomes large enough to hold all the
+            # module headers
+            while True:
+                buf = (HMODULE * buf_count)()
+                buf_size = ctypes.sizeof(buf)
+                if not Psapi.EnumProcessModulesEx(
+                        hProcess, ctypes.byref(buf), buf_size,
+                        ctypes.byref(needed), LIST_MODULES_ALL):
+                    raise OSError('EnumProcessModulesEx failed')
+                if buf_size >= needed.value:
+                    break
+                buf_count = needed.value // (buf_size // buf_count)
+
+            count = needed.value // (buf_size // buf_count)
+            hModules = map(HMODULE, buf[:count])
+
+            # Loop through all the module headers and get the module file name
+            buf = ctypes.create_unicode_buffer(MAX_PATH)
+            nSize = DWORD()
+            for hModule in hModules:
+                if not Psapi.GetModuleFileNameExW(
+                        hProcess, hModule, ctypes.byref(buf),
+                        ctypes.byref(nSize)):
+                    raise OSError('GetModuleFileNameEx failed')
+                module_path = buf.value
+                module_basename = os.path.basename(module_path).lower()
+                if module_basename.startswith(module_name):
+                    found_module_path = module_path
+        finally:
+            Kernel32.CloseHandle(hProcess)
+
+        if found_module_path:
+            return ctypes.CDLL(found_module_path)
+
+    def _get_libc(self):
+        if not hasattr(self, "libc"):
+            libc_name = find_library("c")
+            if libc_name is None:
+                self.libc = None
+            self.libc = ctypes.CDLL(libc_name)
+
+        return self.libc
+
+    def _get_windll(self, dll_name):
+        if not hasattr(self, dll_name):
+            setattr(self, dll_name, ctypes.WinDLL("{}.dll".format(dll_name)))
+
+        return getattr(self, dll_name)
+
+
+_clibs_wrapper = None
+
+
+def _get_wrapper(reload_clib=False):
+    """Helper function to only create one wrapper per thread."""
+    global _clibs_wrapper
+    if _clibs_wrapper is None:
+        _clibs_wrapper = _CLibsWrapper()
+    if reload_clib:
+        _clibs_wrapper._load()
+
+    return _clibs_wrapper
+
+
+def limit_threads_clib(max_threads_per_process, reload_clib=False):
+    """Limit the number of threads available for threadpools in supported C-lib
+
+    Set the maximal number of thread that can be used in thread pools used in
+    the supported C-libraries to `max_threads_per_process`. This function works
+    for libraries that are already loaded in the interpreter and can be changed
```

Comment:
Actually, there is a situation when it will be ignored for the OpenMP libraries. `omp_set_num_threads` does not have the highest priority for fixing the number of threads.

For example if you have a cython function using prange, which looks like 
```for i in prange(num_threads=my_number)```
this will be used for the number of threads. Or more generally, a C program with 
```#pragma omp parallel num_threads(my_number)```

Reply:
I think it's fine to just make a "best effort" and not advertise it as a strict guarantee.

Reply:
right, though I'd say that specifying num_threads for every loop is a bad idea, inherently incomposable. Probably we could influence Cython to remove it or reword in terms of something like omp tasks.